src/api/define.js

```diff
@@ -122,13 +133,16 @@ export default function (name, opts) {
   Ctor[$props] = createInitProps(Ctor);
   Ctor[$renderer] = createRenderer(Ctor);
 
-  if (customElementsV1) {
-    window.customElements.define(uniqueName, Ctor, { extends: Ctor.extends });
-  } else if (customElementsV0) {
-    return document.registerElement(uniqueName, Ctor);
-  } else {
-    throw new Error('Skate requires native custom element support or a polyfill.');
+  let res;
+  try {
```

Comment:
What if we just get rid of the 'registry' object and always return unique name no matter what? This will simplify logic a bit and eliminate try/catch block.
But it will be a breaking change.